api/src/repositories/attractants-repository.ts

```diff
@@ -0,0 +1,165 @@
+        method_technique
+          ON method_technique.method_technique_id = method_technique_attractant.method_technique_id
+      WHERE 
+        method_technique_attractant.method_technique_id = ${methodTechniqueId}
+      AND 
+        method_technique.survey_id = ${surveyId};
+    `;
+
+    const response = await this.connection.sql(sqlStatement, AttractantRecord);
+
+    return response.rows;
+  }
+
+  /**
+   * Create attractant records for a technique.
+   *
+   * @param {number} surveyId
+   * @param {number} methodTechniqueId
+   * @param {IAttractantPostData[]} attractants
+   * @return {*}  {Promise<{ method_technique_attractant_id: number }[]>}
+   * @memberof AttractantRepository
+   */
+  async insertAttractantsForTechnique(
+    surveyId: number,
+    methodTechniqueId: number,
+    attractants: IAttractantPostData[]
+  ): Promise<{ method_technique_attractant_id: number }[]> {
+    if (!attractants.length) {
+      return [];
+    }
+
+    const queryBuilder = getKnex()
+      .insert(
+        attractants.map((attractant) => ({
+          attractant_lookup_id: attractant.attractant_lookup_id,
+          method_technique_id: methodTechniqueId
+        }))
+      )
+      .into('method_technique_attractant')
+      .innerJoin(
+        'method_technique',
+        'method_technique.method_technique_id',
+        'method_technique_attractant.method_technique_id'
+      )
+      .where('method_technique.survey_id', surveyId)
+      .returning('method_technique_attractant_id');
+
+    const response = await this.connection.knex(queryBuilder, z.object({ method_technique_attractant_id: z.number() }));
+
+    return response.rows;
+  }
+
+  /**
+   * Delete technique attractants.
+   *
+   * @param {number} surveyId
+   * @param {number} methodTechniqueId
+   * @param {number[]} attractantLookupIds
+   * @return {*}  {Promise<void>}
+   * @memberof AttractantRepository
+   */
+  async deleteTechniqueAttractants(
+    surveyId: number,
+    methodTechniqueId: number,
+    attractantLookupIds: number[]
+  ): Promise<void> {
+    if (attractantLookupIds.length > 0) {
+      const queryBuilder = getKnex()
+        .delete()
+        .table('method_technique_attractant')
+        .join(
+          'method_technique',
+          'method_technique.method_technique_id',
+          'method_technique_attractant.method_technique_id'
+        )
+        .whereIn('method_technique_attractant.attractant_lookup_id', attractantLookupIds)
+        .andWhere('method_technique_attractant.method_technique_id', methodTechniqueId)
+        .andWhere('method_technique.survey_id', surveyId);
+
+      const response = await this.connection.knex(queryBuilder);
+
+      if (!response.rowCount) {
+        throw new ApiExecuteSQLError('Failed to delete technique', [
+          'AttractantRepository->deleteTechnique',
+          'rows was null or undefined, expected rows != null'
+        ]);
+      }
+    }
+  }
+
+  /**
+   * Delete all technique attractants.
```

Comment:
'Delete all technique attractants of a survey'

Reply:
updated